### PR TITLE
Exchange `imdb_data.zip` and change its host

### PR DESCRIPTION
**Pull request by tjeyy**
Fixes the problem described in #2667.
Alternatively, the `imdb_data` folder could be removed from the zip-file.
What are your opinions on this?

### CODE REVIEW

File: scripts/setup_imdb.py [hunk 1]
```diff
@@ -115,7 +113,7 @@ def is_setup():
 
 hash_dl = hash_md5.hexdigest()
 
-if hash_dl != "24bb992f97dad7b83fd4009e312ddd40":
+if hash_dl != "7c2e84c64126897267d1cf745d47bc9b":
     print("  Aborting. MD5 checksum mismatch. Cleaning up.")
     clean_up()
     sys.exit(2)
```
[the remark this answers is not in the document]
Comment (dey4ss) [reply scripts/setup_imdb.py:119]: True, we could call `is_setup()` after extracting to ensure that.

File: scripts/setup_imdb.py [hunk 2]
```diff
@@ -132,6 +136,11 @@ def is_setup():
     clean_up(including_table_dir=True)
     sys.exit(3)
 
+if not is_setup():
+    print("  Aborting. Unzipping did not result in a correct imdb_files-setup. Cleaning up.")
+    clean_up(including_table_dir=True)
+    sys.exit(3)
```
Comment (dey4ss) [inline scripts/setup_imdb.py:142]:
```suggestion
    sys.exit(4)
```

File: scripts/setup_imdb.py [hunk 3]
```diff
@@ -16,9 +17,14 @@ def clean_up(including_table_dir=False):
...
-            os.remove("./%s/%s" % (table_dir, file))
-        os.rmdir(table_dir)
+        try:
+            shutil.rmtree(table_dir)
+        except PermissionError as e:
+            print(f"PermissionError: Unable to remove {table_dir}. Please check file permissions.")
+            raise
+        except Exception as e:
+            print(f"An error occurred while cleaning up {table_dir}: {e}")
+            raise
```
Comment (dey4ss) [inline scripts/setup_imdb.py:27]: While I like condensing the deletion of the directory into a single call, I am not sure about the value added by catching. I think the errors are telling enough (esp. `PermissionError`). In the other places, we only catch to cleanup (so hopefully re-trying downloads the data again and that already helps), but IMO we can simply crash if the cleanup itself does not work.

**Review by dey4ss — COMMENTED** [review]
Just two small things

File: scripts/setup_imdb.py (hunk 3 above)
Comment (tjeyy) [reply scripts/setup_imdb.py]: Agreed.

**Comment by Bouncner** [comment]
Premature zündung.

**Comment by tjeyy** [comment]
Auto-Merge :/
I'll put the suggested change in my next PR ;)

**Pull request merged**